src/precice/SolverInterface.hpp

```diff
@@ -774,6 +774,75 @@ class SolverInterface {
+   * These API functions are \b experimental and may change in future versions.
+   */
+  ///@{
+
+  /**
+   * @brief setBoundingBoxes Define a region of interest on a received mesh
+   *        (<use-mesh ... from="otherParticipant />") in order to receive
+   *        only a certain mesh region. Have a look at the website under
+   *        https://precice.org/couple-your-code-direct-access.html or
+   *        navigate manually to the page  [sidebar]->Couple your code
+   *        -> Advanced topics -> Accessing received meshes directly for
+   *        a comprehensive documentation
+   *
+   * @experimental
+   *
+   * This function is required if you don't want to use the mapping
+   * schemes in preCICE, but rather want to use your own solver for
+   * data mapping. As opposed to the usual preCICE mapping, only a
+   * single mesh (from the other participant) is now involved in this
+   * situation since an 'own' mesh defined by the participant itself
+   * is not required any more. In order to re-partition the received
+   * mesh, the participant needs to define the mesh region it wants
+   * read data from and write data to.
+   * Defining a bounding box for serial runs of the solver (not to
+   * be confused with serial coupling mode) is valid. However, a
+   * warning is raised in case vertices are filtered out completely
+   * on the receiving side, since the associated data values of the
+   * filtered vertices are filled with zero data.
+   * The mesh region is specified through a collection of axis-aligned
+   * bounding boxes given by the lower and upper [min and max]
+   * bounding-box limits in each space dimension [x, y, z].
+   *
+   * @param[in] meshID ID of the mesh you want to access through the bounding box
+   * @param[in] boundingBoxCollection collection of (axis aligned) bounding boxes which
+   *            has in 3D the format
+   *            [x_min0, x_max0, y_min0, y_max0, z_min0, z_max0, x_min1 ... ]
+   * @param[in] size number of bounding boxes.
+   *
+   * @warning This functions is currently only implemented for nBoundingBoxes = 1.
+   *
+   * @pre 'initialize' has not yet been called.
+   */
+  void setBoundingBoxes(
+      const int     meshID,
+      const double *boundingBoxCollection,
+      const int     size) const;
+
+  /**
+   * @brief getMeshVerticesAndIDs Iterates over the region of
+   *        interest defined by bounding boxes and reads the corresponding
+   *        coordinates omitting the mapping.
+   *
+   * @param[in]  meshID corresponding mesh ID
+   * @param[in]  size return value of getMeshSize
+   * @param[out] ids ids corresponding to the coordinates
+   * @param[out] coordinates associated to the values (dim * @p getLocallyRelevantMeshSize)
+   *
+   * @pre IDs and coordinates need to have the correct size, which can be requested by getMeshVertexSize)
+   * @pre bounding box has been defined using @p setBoundingBoxes
+   */
+  void getMeshVerticesAndIDs(
```

Comment:
> The name doesn't reflect the precondition that the BBs must be set.

That's not the purpose of the name. 

>  The precondition is also very strict. We could simply do nothing if BBs haven't been set yet.

Keeping it strict helps to detect misuse. 

> Also Vertices can be both the coords and the ids, so the AndIDs is not required here.

As you already noted, `vertices` is used for the coordinates in other API functions as well, so that the `AndIDs` makes sense in my opinion. 

> We already have getMeshVertices which returns the coords given ids.
Wouldn't it suffice to add a getMeshMarkedVertexIDs or similar?
Then the user can first get the ids in the bb of the mesh, and then their coordinates.

In my opinion, this makes things for the user more complex. Still open to change it.

Reply:
Indeed, in other API functions we use "vertices" only to refer to the coordinates. So, I think the additional "AndIDs" is good.

About the precondition: do we actually need it? Better would be if this function works for all meshes (which I guess is possible, maybe it even does already, (we would need integration tests for this then though)). Then, there is no room for confusion and less complicated things to document. Do you agree?

Reply:
Agree and I will remove the precondition. Until now, I only considered this API call for this use-case. There might be others.